exporter/elasticsearchexporter/factory.go

```diff
@@ -220,53 +218,21 @@ func exporterhelperOptions(
 	shutdown component.ShutdownFunc,
 	qbs exporterhelper.QueueBatchSettings,
 ) []exporterhelper.Option {
+	// not setting capabilities as they will default to non-mutating but will be updated
+	// by the base-exporter to mutating if batching is enabled.
 	opts := []exporterhelper.Option{
-		exporterhelper.WithCapabilities(consumer.Capabilities{MutatesData: false}),
 		exporterhelper.WithStart(start),
 		exporterhelper.WithShutdown(shutdown),
+		exporterhelper.WithQueueBatch(cfg.QueueBatchConfig, qbs),
 	}
-	qbc := cfg.QueueBatchConfig
-	switch {
-	case qbc.Batch.HasValue():
-		// Latest queue batch settings are used, prioritize them even if sending queue is disabled
-		opts = append(opts, exporterhelper.WithQueueBatch(qbc, qbs))
-
-		// Effectively disable timeout_sender because timeout is enforced in bulk indexer.
-		//
-		// We keep timeout_sender enabled in the async mode (sending_queue not enabled OR sending
-		// queue enabled but batching not enabled OR based on the deprecated batcher setting), to
-		// ensure sending data to the background workers will not block indefinitely.
-		if qbc.Enabled {
-			opts = append(opts, exporterhelper.WithTimeout(exporterhelper.TimeoutConfig{Timeout: 0}))
-		}
-	case cfg.Batcher.enabledSet:
-		if cfg.Batcher.Enabled {
-			qbc.Batch = configoptional.Some(exporterhelper.BatchConfig{
-				FlushTimeout: cfg.Batcher.FlushTimeout,
-				MinSize:      cfg.Batcher.MinSize,
-				MaxSize:      cfg.Batcher.MaxSize,
-				Sizer:        cfg.Batcher.Sizer,
-			})
-
-			// If the deprecated batcher is enabled without a queue, enable blocking queue to replicate the
-			// behavior of the deprecated batcher.
-			if !qbc.Enabled {
-				qbc.Enabled = true
-				qbc.WaitForResult = true
-			}
-		}
 
-		opts = append(
-			opts,
-			// Effectively disable timeout_sender because timeout is enforced in bulk indexer.
-			//
-			// We keep timeout_sender enabled in the async mode (Batcher.Enabled == nil),
-			// to ensure sending data to the background workers will not block indefinitely.
-			exporterhelper.WithTimeout(exporterhelper.TimeoutConfig{Timeout: 0}),
-			exporterhelper.WithQueue(qbc),
-		)
-	default:
-		opts = append(opts, exporterhelper.WithQueue(qbc))
+	// Effectively disable timeout_sender because timeout is enforced in bulk indexer.
+	//
+	// We keep timeout_sender enabled in the async mode (sending_queue not enabled OR sending
+	// queue enabled but batching not enabled OR based on the deprecated batcher setting), to
+	// ensure sending data to the background workers will not block indefinitely.
+	if cfg.QueueBatchConfig.Enabled {
+		opts = append(opts, exporterhelper.WithTimeout(exporterhelper.TimeoutConfig{Timeout: 0}))
 	}
```

Comment:
```suggestion
	opts = append(opts, exporterhelper.WithTimeout(exporterhelper.TimeoutConfig{Timeout: 0}))
```
I believe the `if` can be removed. If queue is enabled, TimeoutSender is after QueueSender and it interferes with our own timeout and retries. If queue is disabled, it is still problematic for the same reason.